lib/AST/ASTDumper.cpp

```diff
@@ -1098,6 +1098,17 @@ namespace {
       PrintWithColorRAII(OS, ParenthesisColor) << ')';
     }
 
+    void visitPoundDiagnosticDecl(PoundDiagnosticDecl *PDD) {
+      const char *name = PDD->isError() ?
+        "pound_error_decl" : "pound_warning_decl";
```

Comment:
Nitpick: seems like having a common `pound_diagnostic_decl` as the kind and then printing an attribute for error vs. warning would be more inline with the other AST nodes.

Reply:
👍